x-py.md

```diff
@@ -0,0 +1,388 @@
+        --build BUILD   build target of the stage0 compiler
+        --host HOST     host targets to build
+        --target TARGET target targets to build
+        --on-fail CMD   command to run on failure
+        --stage N       stage to build
+        --keep-stage N  stage to keep without recompiling
+        --src DIR       path to the root of the rust checkout
+    -j, --jobs JOBS     number of jobs to run in parallel
+    -h, --help          print this help message
+        --test-args ARGS
+                        extra arguments
+```
+
+# Documenting rustc
+
+You might want to build documentation of the various components available like
+the standard library. There's two ways to go about this. You can run `rustdoc`
+directly on the file to make sure the HTML is correct which is fast or you can
+build the documentation as part of the build process through `x.py`. Both are
+viable methods since documentation is more about the content.
+
+### Document everything
+
+```bash
+# Document it all
+./x.py doc
+
+# If you want to avoid the whole Stage 2 build
+./x.py doc --stage 1
+```
+
+First the compiler and rustdoc get built to make sure everything is okay and
+then it documents the files.
+
+### Document specific components
+
+```bash
+./x.py doc src/doc/book
+./x.py doc src/doc/nomicon
+./x.py doc src/doc/book src/libstd
+```
+
+Much like individual tests or building certain components you can build only the
+documentation you want.
+
+### Document internal rustc items
+
+By default `rustc` does not build the compiler for it's internal items.
+Mostly because this is useless for the average user. However, you might need to
+have it available so you can understand the types. Here's how you can compile it
+yourself:
+
+```bash
+cp src/bootstrap/config.toml.example config.toml
+```
+
+Next open up `config.toml` and make sure these two lines are set to true:
+
+```
+docs = true
+compiler-docs = true
+```
+
+When you want to build the compiler docs as well run this command:
+
+```bash
+./x.py doc --config config.toml
+```
+
+This will see that the `docs` and `compiler-docs` options are set to true and
+build the normally hidden compiler docs!
+
+### Other Flags
+
+Like `build`, `doc` also has many optional flags available for it to use:
+
+```
+Options:
+    -v, --verbose       use verbose output (-vv for very verbose)
+    -i, --incremental   use incremental compilation
+        --config FILE   TOML configuration file for build
+        --build BUILD   build target of the stage0 compiler
+        --host HOST     host targets to build
+        --target TARGET target targets to build
+        --on-fail CMD   command to run on failure
+        --stage N       stage to build
+        --keep-stage N  stage to keep without recompiling
+        --src DIR       path to the root of the rust checkout
+    -j, --jobs JOBS     number of jobs to run in parallel
+    -h, --help          print this help message
+```
```

Comment:
Hm, I like this, but also am worried about documentation like this (especially that's out-of-tree) going stale. I guess we'll keep it for now though...